src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/induction/ParentEntity.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction
+
+/**
+ * Abstract class for "parent" entities, which contain one or more collections of [KeyAwareChildEntity] entities.
+ */
+abstract class ParentEntity {
+
+  abstract fun childEntityUpdated()
+
+  fun <CHILD : KeyAwareChildEntity> addChild(newChild: CHILD, existingChildren: MutableList<CHILD>) {
+    addChildren(mutableListOf(newChild), existingChildren)
+  }
+
+  fun <CHILD : KeyAwareChildEntity> addChildren(newChildren: List<CHILD>, existingChildren: MutableList<CHILD>) {
+    newChildren.forEach {
+      it.associateWithParent(this)
```

Comment:
As noted above, this is where the link is made, when adding a child entity to the parent's list.